Refactor access derivative script

kfrn opened the pull request:
I'm on a roll! Some more changes 😀

Comments inline below.

## Problem!
I just noticed that my test files are being created without audio streams. I haven't changed anything to do with the encoding itself, really, so I'm wondering if this was an existing problem - @digitensions, are the files arriving with audio for you?

kfrn committed:
Refactor: early return for invalid files; indicate unused var; naming

kfrn committed:
Add function for setting output path + giving default

If your inout file path is path/to/my-file.mp4, the default output path will be path/to/my-file_access_copy.mp4.

kfrn committed:
Refactor: simplify `display_aspect_ratio` & `stream_aspect_ratio`

`aspect_ratio_and_resolution` → `stream_aspect_ratio`
We never actually use the resolution, so no need to return it.

kfrn committed:
Fix: add missing output settings for watermarked + untrimmed output

kfrn committed:
Refactor: store options (trim/watermark) as Boolean flags

Option setting extracted into separate methods

kfrn commented on main.py:
```diff
@@ -50,45 +50,30 @@ def main():
 
+        if not (os.path.isfile(input_file)):
+            print("ERROR: Path not valid.")
+
```
Not a real change! I just moved this up out of personal preference

digitensions replied:
No audio! Woooo I need to check. I think my test files have all been silent. Thanks for spotting this. -c:a is in the subprocess call of the second version of this script main_new.py so it should be working fine. I'll check this end - you wanna try some files on the subprocess call version main_new.py?

Many thanks @kfrn this is soooo much fun :)

kfrn commented on main.py:
```diff
@@ -50,45 +50,30 @@ def main():
-                    break
+        else:
+            to_trim = set_trim_status()
+            to_watermark = set_watermark_status()
```
I think it's cleaner to work with these as Booleans rather than storing `"y"`/`"n"` strings

kfrn commented on main.py:
```diff
@@ -50,45 +50,30 @@ def main():
-            # Get aspect ratio, resolution, and watermark image.
-            aspect_ratio, resolution = aspect_ratio_and_resolution(input_file)
+            # Get aspect ratio and watermark image.
+            aspect_ratio = stream_aspect_ratio(input_file)
```
The `resolution` is never actually used, so I refactored the function not to return it.

It's good practice for a function to have a single purpose, where practical - a function with an `and` (actual or implicit) in the name is a sign that it needs to be refactored, IMO. (I might have broken this out into two functions if we actually needed `resolution`).    
(apologies if I'm stating the obvious here!)

digitensions replied:
No, please do. Most of this is James' work which I'm struggling to understand. MANY THANKS!! :)

kfrn commented on main.py:
```diff
@@ -128,42 +113,42 @@ def main():
+    width = int(numbers[0])
+    height = int(numbers[1])
 
+    return width / height
```
Not really a necessary change, but for me easier to read than before - what do you think?

kfrn commented on main.py:
```diff
@@ -128,42 +113,42 @@ def main():
     # Function to get a path to a corresponding .png watermark given an aspect ratio string.
 
-    for name, data in RESOLUTIONS.items():
+    for name, _ in RESOLUTIONS.items():
```
This is iterating over the `RESOLUTIONS` dict, i.e. `for key, value in RESOLUTIONS` - but since we only care about the keys, I've named the `value` param `_`, to indicate that it's unused.

kfrn commented on main.py:
```diff
@@ -172,17 +157,55 @@ def watermark_path(ratio):
+
+    if not (trim == "y" or trim == "n"):
+        print("Invalid input! I'll ask again")
+        set_trim_status()
```
Using recursion here, as already for `set_watermark_status` & `set_timestamp`, and below for `set_watermark_status`. How do you find this behaviour? You may wish for different 😉

kfrn commented:
Yeah, let me know about the audio! 🔕And I'll give `main_new` a whirl when I'm at home later. **Edit**: yep, audio present & correct with `main_new`.

Quite bizarre about the vanishing audio. I didn't have the chance to debug it, but a glance, I don't see much about stream mapping in the [`ffmpeg-python`](https://github.com/kkroening/ffmpeg-python) docs. They [say](https://kkroening.github.io/ffmpeg-python/#ffmpeg.output) re: output,
> If multiple streams are provided, they are mapped to the same output.

😐

If I understood correctly, `main_new` is the same functionality as `main`, just using a subprocess call rather than using the Python ffmpeg wrapper, right? As such, I wonder if the next step is to consolidate the two scripts by refactoring `main` to use the subprocess calls.

You're welcome, I'm enjoying the process! 😊

kfrn commented:
Confirming that the changes in this PR don't affect the audio stream - in other words, the script was already producing mute files. No big deal though, we can fix that in a follow-up!

kfrn committed:
Docs: update info on main.py (access derivative creation script)

Update script prompts to current wording, clarify dependencies, and improve general readability
